src/server/node_manager.c

```diff
@@ -2275,10 +2275,103 @@ stat_update(int stream)
 			(pjob->ji_wattr[(int)JOB_ATR_run_version].at_val.at_long == rused.ru_hop)) {
 
 			long old_sid = 0;  /* used to save prior sid of job */
+			svrattrl	*execvnode_entry = NULL;
+			svrattrl	*schedselect_entry = NULL;
+			char		*cur_execvnode = NULL;
+			char		*cur_schedselect = NULL;
 
+			if (pjob->ji_wattr[(int)JOB_ATR_exec_vnode].at_flags & ATR_VFLAG_SET)
+				cur_execvnode = pjob->ji_wattr[(int)JOB_ATR_exec_vnode].at_val.at_str;
+
+			if (pjob->ji_wattr[(int)JOB_ATR_SchedSelect].at_flags & ATR_VFLAG_SET)
+				cur_schedselect = pjob->ji_wattr[(int)JOB_ATR_SchedSelect].at_val.at_str;
+
+			/* update all the attributes sent from Mom */
+			execvnode_entry = find_svrattrl_list_entry(&rused.ru_attr, ATTR_execvnode, NULL);
+			schedselect_entry = find_svrattrl_list_entry(&rused.ru_attr, ATTR_SchedSelect, NULL);
+
+			if ((execvnode_entry != NULL) &&
+			    (execvnode_entry->al_value != NULL) &&
+			    (schedselect_entry != NULL) &&
+			    (schedselect_entry->al_value != NULL) &&
+			    (cur_execvnode != NULL) &&
+			    (strcmp(cur_execvnode, execvnode_entry->al_value) != 0) &&
+			    (cur_schedselect != NULL) && 
+			    (strcmp(cur_schedselect, schedselect_entry->al_value) != 0)) {
+
+				/* decreements everything found in exec_vnode */
+				set_resc_assigned((void *)pjob, 0, DECR);
+				free_nodes(pjob);
+
+				if (cur_execvnode != NULL) {
+					(void)job_attr_def[(int)JOB_ATR_exec_vnode_acct].at_decode( &pjob->ji_wattr[(int)JOB_ATR_exec_vnode_acct], (char *)0, (char *)0, cur_execvnode);
+				}
+
+				if ((pjob->ji_wattr[JOB_ATR_resource_acct].at_flags & ATR_VFLAG_SET) != 0) {
+					job_attr_def[JOB_ATR_resource_acct].at_free(&pjob->ji_wattr[JOB_ATR_resource_acct]);
+					pjob->ji_wattr[JOB_ATR_resource_acct].at_flags &= ~ATR_VFLAG_SET;
+				}
+				job_attr_def[JOB_ATR_resource_acct].at_set(&pjob->ji_wattr[JOB_ATR_resource_acct], &pjob->ji_wattr[JOB_ATR_resource], INCR);
+
+
+				(void)job_attr_def[(int)JOB_ATR_exec_host_acct].at_decode(
+					&pjob->ji_wattr[(int)JOB_ATR_exec_host_acct],
+					(char *)0,
+					(char *)0,
```

Comment:
Shouldn't we just pass NULL instead of (char *)0? (multiple instances, just flagging this one)

Reply:
Yeah, just NULL will be fine. I've made the change to this and other instances.